spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java

```diff
@@ -117,7 +118,9 @@ public Resource getResource(String location) {
 				return resource;
 			}
 			else {
-				String cacheName = scheme + "-" + ShaUtils.sha1(location) + "-" + resource.getFilename();
+				String fileName = resource.getFilename();
```

Comment:
Frankly, having had a second look at the code, why do we even bother with the filename.
The `location` already includes the filename (in some way or the other) and is the perfect cache key.

`sha1(scheme + location)` or `scheme + sha1(location)` is the simplest and more sensible solution.

I can change on merge